src/target/adiv5_swd.c

```diff
@@ -53,10 +53,66 @@ uint8_t make_packet_request(uint8_t RnW, uint16_t addr)
 
 /* Provide bare DP access functions without timeout and exception */
 
-static void swd_line_reset(void)
+static void swd_line_reset_sequence(const bool idle_cycles)
 {
-	swd_proc.seq_out(0xffffffffU, 32U);
-	swd_proc.seq_out(0x0fffffffU, 32U);
+	/* 
+	 * A line reset is achieved by holding the SWDIOTMS HIGH for at least 50 SWCLKTCK cycles, followed by at least two idle cycles
+	 * Note: in some non-conformant devices (STM32) at least 51 HIGH cycles and/or 3/4 idle cycles are required
+	 *
+	 * for robustness, we use 60 HIGH cycles and 4 idle cycles
+	 */
+	swd_proc.seq_out(0xffffffffU, 32U);                     /* 32 cycles HIGH */
+	swd_proc.seq_out(0x0fffffffU, idle_cycles ? 32U : 28U); /* 28 cycles HIGH + 4 idle cycles if idle is requested */
+}
+
+/* Switch out of dormant state into SWD */
+static void dormant_state_to_swd_sequence()
```

Comment:
We know this isn't technically required because of that `static`, but to be in keeping with enforcing proper prefixing, we suggest making this `swd_switch_from_dormant_sequence()` - either that or drop the word state, making this `dormant_to_swd_sequence()`